src/Disks/DiskBackup.cpp

```diff
@@ -0,0 +1,75 @@
+#include "DiskBackup.h"
+
+#include <Disks/DiskFactory.h>
+#include "Disks/IDisk.h"
+#include <Disks/ObjectStorages/MetadataStorageFactory.h>
+
+
+namespace DB
+{
+
+DiskBackup::DiskBackup(const String & name_, DiskPtr delegate_, MetadataStoragePtr metadata_) : IDisk(name_), delegate(delegate_), metadata(metadata_)
+{
+
+}
+
+DiskBackup::DiskBackup(const String & name_, const Poco::Util::AbstractConfiguration & config_, const String & config_prefix_, const DisksMap & map_) : IDisk(name_)
+{
+    String disk_delegate_name = config_.getString(config_prefix_ + ".disk_delegate");
```

Comment:
May be we can rename `disk_delegate` to just `disk`, because in some similar cases we use just "disk", for example for disk type "cache".